log4j-api/src/main/java/org/apache/logging/log4j/internal/DefaultLogBuilder.java

```diff
@@ -246,16 +231,15 @@ private void logMessage(final Message message) {
         try {
             logger.logMessage(level, marker, fqcn, location, message, throwable);
         } finally {
-            inUse = false;
+            // recycle self
+            this.level = null;
+            this.marker = null;
+            this.throwable = null;
+            this.location = null;
```

Comment:
I fail to understand how can the recycler know that the `LogBuilder` can be used again?